### ISSUE TITLE
Emacs freezes (100% CPU) on `helm-ag` and `helm-do-ag`

### ISSUE THREAD

**Issue by bendiksolheim** [syohex/emacs-helm-ag#308]
- Emacs version: 26.0.90 (9.0)
- Operating system: macOS Sierra (10.12.6)
- Search command(ag, pt, ack etc) and version: ag 2.1.0
- Last value of `helm-ag--last-command`: nil

### Actual behavior
Emacs freezes whenever I use `helm-do-ag` or `helm-ag` and input more than 2 characters

### Expected behavior
Emacs should not freeze

### Steps to reproduce
- Open Emacs
- Open any file
- Run `helm-do-grep`
- Start writing. Emacs now freezes on my machine (and constantly uses 99% CPU)

### More info
This seems to be a problem with "newer" versions of Emacs. I have not managed to pinpoint exactly which versions, but I have problems with several of the distributions (emacsformacosx, railwaycat/homebrew-emacsmacport and d12frosted/homebrew-emacs-plus at least). The railwaycat-port is the only one that works, but I get the exact same problem if I install with the `--HEAD` option (which, I know, is risky at best).

I have tried switching `helm-ag-base-command` to be `pt -e --nocolor --nogroup`, and I have the exact same behavior as with `ag`, so I do not think this is an issue with `ag` itself. Also, `ag.el` works as intended.

I have tried debugging this quite a lot, but I have to admit defeat. I have absolutely no clue as to why this happens. I am happy to test things and do some debugging, but I would need some input on where to look. I have tried debugging with `debug-on-error` and `debug-on-quit`, but because Emacs freezes, this does not seem to work. Pressing `C-g` does not work. Actually, quitting Emacs (with `cmd-q`) does sometimes trigger a dialog asking me if I want to quit even though subprocesses are still running, but pressing `y` here does not do anything. The only way to actually quit Emacs when this problem arises is killing the Emacs process.

**Comment by RockyRoad29** [comment]
Something similar happened to me a few times in recent weeks. As you say it hard to investigate. Today it froze during flyspell auto-completion. I'm on spacemacs which uses a lot of helm automatically.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.3.1
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 723604d90)
- Graphic display: t
- Distribution: spacemacs
- Editing style: vim
- Completion: helm
- Layers:
```elisp
(helm auto-completion csv dash emacs-lisp git github graphviz
      (gtags :variables gtags-enable-by-default nil)
      html jabber javascript latex markdown neotree
      (org :variables org-enable-github-support t org-enable-reveal-js-support t)
      pandoc perl5 php plantuml python restclient
      (spell-checking :variables spell-checking-enabled-by-default nil)
      sql syntax-checking version-control yaml rr-chatting-windows)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11 MODULES

**Comment by bendiksolheim** [comment]
I believe I have found the culprit of this now. It seems that `helm` has (sometimes, and not always? I am not sure) problems with ligatures according to https://github.com/tonsky/FiraCode/wiki/Emacs-instructions and https://github.com/tonsky/FiraCode/issues/158 . I am not sure why it just sometimes happens, or why it only happens to some ligatures. But as I don’t really see the need for ligatures in `helm`, I disabled them in the helm buffer with the provided workaround:
```
   (add-hook 'helm-major-mode-hook
          (lambda ()
            (setq auto-composition-mode nil)))
```
and Emacs no longer freezes when I use `helm-ag`. I consider my problem fixed, and will therefore close this issue. It also does not seem to have anything to do with this package, so keeping it open is of no use.
